nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -191,18 +192,18 @@ trait NirGenExpr { self: NirGenPhase =>
       val mergev               = Val.Local(fresh(), retty)
 
       val cond = genExpr(condp)
-      buf.branch(cond, Next(thenn), Next(elsen))
+      buf.branch(cond, Next(thenn), Next(elsen))(condp.pos)
       locally {
-        buf.label(thenn)
+        buf.label(thenn)(thenp.pos)
         val thenv = genExpr(thenp)
-        buf.jump(mergen, Seq(thenv))
+        buf.jump(mergen, Seq(thenv))(NoPosition)
```

Comment:
IMO there should be virtually no reason to generate `NoPosition` from the compiler back-end, ever. Even for code that seems "generated", you should always take the position of the node that caused the generated instruction to appear. In this case for example, it should be the `pos` of the `If` node.